### DISCUSSION TITLE
Function to Disable QR Code Processing

### DISCUSSION THREAD

**Discussion by ipv6freely** [gopro/labs#290] [Feature Requests]
I've looked through the documentation and haven't really seen a way to temporarily disable QR code processing (while still maintaining the ability to read QR codes). If I missed something, please feel free to link me and post an ACKCHYUALLY meme in response.

That said, this is kind of what I was thinking: Implement a function that will accept a QR code with a password that tells the camera to ignore all subsequent QR codes it sees until the same QR code again. Pretty simple, intuitive, and non-intrusive. Just a quick lock mechanism without having to disable QR code reading altogether.

**Comment by dnewman-gpsw** [comment]
The closest is the concept of a handle, see HNDL under https://gopro.github.io/labs/control/extensions/

This gives you camera an ID, and only QR codes with matching ID mask will act on the decoded data.

Info from the extensions page:
!M**HNDL**=x, where x is 1 to 31, setting the camera ID for a camera. This is for rare scenarios where multiple cameras see the same QR Code, and you only want particular cameras to respond. This combined with hZ command where Z is the bit mask for which cameras will follow the command.
e.g. h6mP!S ← this command will only run on cameras with IDs 2 and 3.
e.g. h1mVh2mPB ← set camera 1 to mode Video and camera 2 to Photo Burst.

**Reply by ipv6freely** [reply]
Okay, good to know I didn't miss anything in the docs! I saw handle, but that's not quite what I'm looking for. Still way too easily exploitable.

**Reply by dnewman-gpsw** [reply]
Proposed a command set, that would work within the current system.

**Reply by ipv6freely** [reply]
Sure, something very simple like `oMPASS=s3cr37P@55w0rd` that just tells the camera that all subsequent QR codes are to be ignored until it once again reads a QR code that contains `oMPASS=s3cr37P@55w0rd`. 

Example: https://gopro.github.io/labs/control/set/?cmd=oMPASS=s3cr37P@55w0rd

**Reply by dnewman-gpsw** [reply]
Do you use the same command twice?   That will likely not work, as you can too easily scan a code twice.  Plus you added a new security problem, users can control other cameras with a QR Code, and locking the owner out, so worse than now. This stuff is not easy, that is one reason it hasn't been done.  There needs to be a complete design, and I'm only a security hobbyist.

**Reply by ipv6freely** [reply]
Fair enough, yeah it will definitely need to be thought through. But its definitely needed in some way shape or form :)

**Reply by dnewman-gpsw** [reply]
I've never heard of a QR code hijack yet on GoPro Labs.  So the need is not great today, but as this becomes more popular, we will have to add more security, reducing ease of use.

**Reply by ipv6freely** [reply]
For sure. Thanks for the discussion, it was just something that I thought of while I was using it for the first time. I'll go ahead and close the discussion.
